content/guides/styling-your-components.mdx

```diff
@@ -0,0 +1,522 @@
+The `.scss`  syntax is a strict superset of CSS, while `.sass`  offers a more relaxed syntax.
+Vite, which is integrated with Solid, supports both. 
+However, `.scss`  is generally recommended.
+
+```scss
+// Card.scss
+.grid {
+  display: grid;
+  &.center { place-items: center; }
+}
+.screen { min-height: 100vh; }
+.card {
+  height: 160px;
+  aspect-ratio: 2;
+  border-radius: 16px;
+  background-color: white;
+  box-shadow: 0 0 0 4px hsl(0 0% 0% / 15%);
+}
+```
+
+In a Solid component:
+
+```jsx
+// Card.jsx
+import "./card.scss"
+
+function Card() {
+  return (
+    <>
+      <div class="grid grid-center screen">
+        <div class="card">Hello, world!</div>
+      </div>
+    </>
+  );
+}
+```
+
+By simply changing the file extension from `.css`  to `.scss`  or `.sass` , Vite will automatically recognize these files and compile SASS to CSS on demand. 
+When building in production, all SASS files are converted to CSS, to ensure compatibility with browsers.
+
+### LESS
+
+[LESS](https://lesscss.org/) is a preprocessor based on JavaScript. 
+It provides the ability to use mixins, variables, and other programmatic tools, making styling code cleaner and less redundant.
+
+#### Installation
+
+To utilize LESS in a Solid app, it will need to be installed as a development dependency:
+
+```bash
+npm install --save-dev less
+
+# or
+
+pnpm i --dev less
+
+# or
+
+yarn add --dev less
+```
+
+#### Using LESS in your app
+
+Start by creating a `.less`  file in the `src` directory:
+
+```less
+//styles.less
+.foo {
+  color: red;
+}
+.bar {
+  background-color: blue;
+}
+```
+
+The basic syntax of LESS is very similar to CSS.
+However, LESS allows the declaration and usage of variables:
+
+```less
+//styles.less
+@plainred: red;
+@plainblue: blue;
+.foo {
+  color: @plainred;
+}
+.bar {
+  background-color: @plainblue;
+}
+```
+
+To use these styles in a Solid component, import the `.less` file:
+
+```jsx
+//component.jsx
+import "./styles.less";
+
+function Component() {
+  return (
+    <>
+      <div class="foo bar">Hello, world!</div>
+    </>
+  );
+}
+```
+
+By changing the file extension of the imported styles to `.less`, Vite will recognize it as a LESS file and compile it to CSS on demand.
+
+## CSS modules
+
+CSS Modules are CSS files where class names, animations, and media queries are scoped locally by default. 
+These provide a way to encapsulate styles within components, preventing global conflicts and optimizing the final output by bundling only the used selectors.
+
+### Creating CSS module files
+
+Begin by creating a CSS module file. 
+Conventionally, these files have a `.module.css` extension, like `style.module.css`. 
+However, other other extensions can also be used, such as `.scss` and `.sass`.
+
+```css
+/* styles.module.css */
+.foo {
+  color: red;
+}
+.bar {
+  background-color: blue;
+}
+``` 
+**Note:** Avoid the use of HTML tags in CSS modules, as they are not considered pure selectors.
```

Comment:
Would it be worth a short sentence explaining why this is a bad thing? Or is this advanced enough of a document that it is assumed that the audience knows the consequences of this?